src/gtest/test_blockdownload.cpp

```diff
@@ -0,0 +1,95 @@
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "main.h"
+
+using namespace zen;
+
+void makeTemporaryMainChain(int trunk_size);
+
+// Checks IsInitialBlockDownload() conditions and lockIBDState latching to false
+TEST(initialblockdwnld, checkIBDState) {
+    // Init
+    auto originalTip = chainActive.Tip();
+    chainActive.SetTip(NULL);
```

Comment:
I'd suggest to use `nullptr` consistently for the new code that we write, as you do in the rest of the function

Reply:
changed to nullptr in commit 8036ea80c617507a1a3fd033f29bfeaa13b08ff0